src/mscorlib/src/System/Runtime/CompilerServices/AsyncMethodBuilder.cs

```diff
@@ -575,10 +446,10 @@ public void SetResult(TResult result)
         {
             // Get the currently stored task, which will be non-null if get_Task has already been accessed.
```

Comment:
If you zero it out, that could impede any "interesting" state captured by the state machine in the debugger, right? Is that worth keeping around? Esp if an Exception happens, couldn't the state machine state could help?